### ISSUE TITLE
Compile Release 1.5.17 Error: emojier.vala:228.43-228.56: error: The type name `IBus.EmojiData' could not be found

### ISSUE THREAD

**Issue by huadong** [ibus/ibus#1953]
VALAC    ibus_ui_gtk3_vala.stamp
emojier.vala:228.43-228.56: error: The type name `IBus.EmojiData' could not be found
    private static GLib.HashTable<string, IBus.EmojiData>?

**Comment by huadong** [comment]
try to change the option from **link_all_deplibs=no** to **link_all_deplibs=yes** in ./libtool

error: Package `ibus-1.0' not found in specified Vala API directories or GObject-Introspection GIR 

> make[3]: Entering directory '/home/hdzhong/workspace-c/ibus/ui/gtk3'
> rm -f ibus_ui_gtk3_vala.stamp && echo stamp > ibus_ui_gtk3_vala.stamp-t
> CDPATH="${ZSH_VERSION+.}:" && cd . && /usr/bin/valac --vapidir=../../bindings/vala --vapidir=../../bindings/vala --pkg=posix --pkg=gtk+-3.0 --pkg=gdk-x11-3.0 --pkg=ibus-1.0 --pkg=config --pkg=xi --target-glib="2.36"  --pkg=libnotify -D ENABLE_LIBNOTIFY  --define=INDICATOR  --define=EMOJI_DICT  -C application.vala candidatearea.vala candidatepanel.vala emojier.vala handle.vala iconwidget.vala indicator.vala keybindingmanager.vala panel.vala pango.vala property.vala propertypanel.vala separator.vala switcher.vala xkblayout.vala
> error: Package `ibus-1.0' not found in specified Vala API directories or GObject-Introspection GIR directories
> Compilation failed: 1 error(s), 0 warning(s)
> Makefile:855: recipe for target 'ibus_ui_gtk3_vala.stamp' failed

` --pkg=ibus-1.0` ??

**Comment by fujiwarat** [comment]
I think you don't have to change link_all_deplibs.
IBus.EmojiData is defined in $IBUS_BUILD/src/IBus-1.0.gir and it's converted to $IBUS_BUILD/bindings/vala/ibus-1.0.vapi and ibus-1.0.vapi includes IBus.EmojiData
```
% grep EmojiData ibus-1.5.17/bindings/vala/ibus-1.0.vapi 
	public class EmojiData : IBus.Serializable {
```

**Comment by huadong** [comment]
Fixed after install **libgirepository1.0-dev**. It should be in system requirements?

It also fix #1952 .

Thank you for your help! @fujiwarat
